tinygrad/ops.py

```diff
@@ -892,80 +892,63 @@ def split_uop(x:UOp, sep:Ops):
     for s in x.src: yield from split_uop(s, sep)
   else: yield x
 
-def mod_folding(x:UOp, c:int) -> Optional[UOp]:
-  # simplify x % c, None means no change
+def div_and_mod_folding(x: UOp, c: int, which: Literal[Ops.MOD, Ops.IDIV], split_rem: bool=False) -> Optional[UOp]:
+  # simplify x // c or x % c, None means no change, c must be > 0
+  assert c > 0
```

Comment:
if we are asserting this, i think it's good to also assert which is either `Ops.MOD` or `Ops.IDIV`

Reply:
I asserted `c>0` because it would not be a type error to input a negative `c`,  `which` is fully typed though so that should be caught by mypy